pkg/odo/cli/component/create.go

```diff
@@ -124,6 +128,14 @@ A full list of component types that can be deployed is available using: 'odo cat
 			os.Exit(1)
 		}
 
+		// if --git is not specified but --ref is still given then error has to be thrown
+		if len(componentGit) == 0 {
+			if len(componentGitRef) != 0 {
```

Comment:
This looks odd..

Shouldn't it be `len(componentGit) == 0 && len(componentGitRef) != 0`

Reply:
ack